conda/activate.py

```diff
@@ -75,48 +72,45 @@ def get_export_unset_vars(self, odargs):
                        .. if you pass and set any other variable to None, then it
                        emits it to the dict with a value of None.
 
-        :return: A OrderedDict of env vars to export ordered the same way as kwargs.
+        :return: A dict of env vars to export ordered the same way as kwargs.
                  And a list of env vars to unset.
         """
-        kwargs = odargs
-        conda_exe_vars_None = False if ('conda_exe_vars' not in kwargs or
-                                        kwargs['conda_exe_vars'] is not None) else True
-        conda_exe_unset_vars = []
         unset_vars = []
-        # conda_exe_vars = context.conda_exe_vars_dict.copy()
-        from collections import OrderedDict
-        conda_exe_vars_export = OrderedDict()
-        for k, v in context.conda_exe_vars_dict.items():
-            if v is None or conda_exe_vars_None:
-                conda_exe_unset_vars.append(k)
-            else:
-                conda_exe_vars_export[k] = self.path_conversion(v) if v else v
-        from collections import OrderedDict
-        export_vars = OrderedDict()
-        for k, v in kwargs.items():
-            if k == 'conda_exe_vars':
-                continue
-            elif v is None:
-                unset_vars.append(k.upper())
-            else:
-                export_vars[k.upper()] = v
-        # Just to make writing tests a bit more ergonomic we add these to the end.
-        unset_vars += conda_exe_unset_vars
-        export_vars = OrderedDict(chain(export_vars.items(), conda_exe_vars_export.items()))
+        export_vars = {}
+
+        def split_export_unset(func=None, **kwargs):
```

Comment:
Perhaps it would be better not to shadow the already existing `kwargs` variable. I usually go with `**kwargs_` (add a tail) or `**inner_kwargs` to avoid this.

Reply:
Hm, personally I've never been a fan of using alternatives for `args` or `kwargs` unless I absolutely must (e.g., for decorators).

I'm also not a fan of the proposed implementation, what I really want to do was:

```python
def get_export_unset_vars(self, **kwargs):
    unset_vars = []
    export_vars = {}

    for name, value in {**kwargs, **context.conda_exe_vars_dict}.items():
        if value is None:
            unset_vars.append(name.upper())
        else:
            export_vars[name.upper()] = value
```

But things got weird with the `conda_exe_vars` conditional and the `path_conversion` only being applied to the `context.conda_exe_vars_dict` (and not to `kwargs`).

I'll simmer on this a little longer. 🤔

Reply:
I decided to drop the helper function since it didn't help with the overall code readability